docs/osgi.md

```diff
@@ -0,0 +1,38 @@
+# OSGI background information
+
+Some (rather incomplete) information about bundles, bundle dependencies and errors in that context.
+
+## METAINF.MF
+### Require-Bundle
+
+- specifies the explicit bundle (and optionally version) to use. If a required bundle needs to be refactored and a package moved elsewhere, then dependents will need changes to their MANIFEST.MF
+- gives you access to ALL exports of the bundle, regardless of what they are, and regardless of whether you need them. If the parts you don't need have their own dependencies you will need those to
+- bundles can be re-exported
+- although discouraged, allows the use of split packages, e.g.: a package that is spread across multiple bundles
+- can be used for non-code dependencies, e.g.: resources, Help etc.
+
+### Import-Package (preferred)
```

Comment:
Importing packages should not be preferred. I think it's only okay for a certain set of third-party dependencies.
If you start importing your own packages (of other bundles) this will get very messy.